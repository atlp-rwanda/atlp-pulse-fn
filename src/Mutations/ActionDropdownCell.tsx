import React from 'react';
import ActionDropdown from '../components/ActionDropdown';

interface ActionDropdownCellProps {
  row: {
    original: any;
  };
  onView: (ticket: any) => void;
  onEdit?: (id: string) => void;
  onDelete?: (id: string) => void;
  canEditDelete: boolean;
}

function ActionDropdownCell({
  row,
  onView,
  onEdit,
  onDelete,
  canEditDelete,
}: ActionDropdownCellProps) {
  return (
    <ActionDropdown
      onView={() => onView(row.original)}
      onEdit={canEditDelete ? () => onEdit?.(row.original.id) : undefined}
      onDelete={canEditDelete ? () => onDelete?.(row.original.id) : undefined}
      canEditDelete={canEditDelete}
      id={row.original.id}
    />
  );
}

export default ActionDropdownCell;
