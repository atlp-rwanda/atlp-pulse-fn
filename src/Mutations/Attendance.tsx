import { gql } from '@apollo/client';

export const GET_ATTENDANCE = gql`
  query ExampleQuery {
    getTraineeAttendance {
      id
      trainees {
        traineeId
        traineeEmail
        status {
          days
          value
        }
      }
      week
    }
  }
`;

export const GET_ATTENDANCE_BY_ID = gql`
  query GetAttendance($id: ID!) {
    getTraineeAttendanceByID(id: $id) {
      id
      week
      trainees {
        traineeId
        traineeEmail
        status {
          days
          value
        }
      }
    }
  }
`;

export const GET_WEEKLY_ATTENDANCE = gql`
  query GetTraineeAttendanceByID($traineeEmail: String!) {
    getTraineeAttendanceByID(traineeEmail: $traineeEmail) {
      weekNumber
      traineeAttendance {
        days
        value
      }
    }
  }
`;

export const UPDATE_ATTENDANCE = gql`
  mutation RecordAttendance(
    $week: String!
    $days: String!
    $trainees: [TraineeInput!]!
    $recordAttendanceOrgToken2: String!
  ) {
    recordAttendance(
      week: $week
      days: $days
      trainees: $trainees
      orgToken: $recordAttendanceOrgToken2
    ) {
      id
      week
      trainees {
        traineeId
        traineeEmail
        status {
          days
          value
        }
      }
    }
  }
`;
