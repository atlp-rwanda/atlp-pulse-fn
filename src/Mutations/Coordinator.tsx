import { gql } from '@apollo/client';


export const GetAllCoordinatorUsers = gql`
  query GetCoordinatorUsers($orgToken: String) {
    getAllCoordinatorUsers(orgToken: $orgToken) {
      id
      email
      profile {
        name
      }
    }
  }
  
`;