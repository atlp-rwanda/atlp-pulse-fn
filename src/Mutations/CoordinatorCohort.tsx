import { gql } from '@apollo/client';

export const GetCoordinatorCohort = gql`
  query GetCoordinatorCohort($coordinatorId: String) {
    getCoordinatorCohort(coordinator_id: $coordinatorId) {
      program {
        name
      }
      name
    }
  }
`;
