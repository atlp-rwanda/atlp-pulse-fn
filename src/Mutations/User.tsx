import { gql } from '@apollo/client';

export const GET_PROFILE = gql`
  query {
    getProfile {
      firstName
      lastName
      phoneNumber
      address
      city
      country
      avatar
      cover
      name
      biography
      githubUsername
      user {
        organizations
        email
        role
        team {
          name
          cohort {
            name
            phase {
              name
            }
            program {
              name
            }
            startDate
          }
        }
        cohort {
          name
          program {
            name
          }
          phase {
            name
          }
        }
        program {
          name
        }
      }
    }
  }
`;

export const GET_ALL_TTL_USERS = gql`
  query GetAllTTLUsers($orgToken: String!) {
    getAllTTLUsers(orgToken: $orgToken) {
      profile {
        name
      }
      email
      team {
        name
        cohort {
          name
        }
      }
    }
  }
`;

export const GET_TTL_TRAINEES = gql`
  query GetTTLTrainees($orgToken: String) {
    getTTLTrainees(orgToken: $orgToken) {
      profile {
        githubUsername
        address
        name
        avatar
        resume
      }
      email
      role
      team {
        name
        cohort{
          name
        }
      }
      cohort {
        name
      }
      program {
        name
        manager {
          email
        }
      }
    }
  }
`;
export const DROP_TTL_USER = gql`
  mutation DropTTLUser($email: String!, $reason: String!) {
    dropTTLUser(email: $email, reason: $reason)
  }
`;
export default GET_PROFILE;
