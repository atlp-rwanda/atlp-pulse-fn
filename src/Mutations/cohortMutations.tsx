import { gql } from '@apollo/client';

export const AddCohort = gql`
  mutation AddCohort(
    $name: String!
    $phaseName: String!
    $coordinatorEmail: String!
    $programName: String!
    $startDate: DateTime!
    $endDate: DateTime
    $orgToken: String!
  ) {
    addCohort(
      name: $name
      phaseName: $phaseName
      coordinatorEmail: $coordinatorEmail
      programName: $programName
      startDate: $startDate
      endDate: $endDate
      orgToken: $orgToken
    ) {
      id
      name
      phase {
        name
      }
      coordinator {
        email
      }
      program {
        name
      }
      startDate
      endDate
    }
  }
`;

export const DeleteCohort = gql`
  mutation DeleteCohort($deleteCohortId: ID!, $orgToken: String) {
    deleteCohort(id: $deleteCohortId, orgToken: $orgToken) {
      id
    }
  }
`;
