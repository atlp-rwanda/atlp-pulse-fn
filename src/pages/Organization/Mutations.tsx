import { gql } from '@apollo/client';

export const SIGN_UP_MUTATION = gql`
  mutation Mutation(
    $firstName: String!
    $lastName: String!
    $dateOfBirth: DateTime!
    $gender: String!
    $email: String!
    $password: String!
    $orgToken: String!
  ) {
    createUser(
      firstName: $firstName
      lastName: $lastName
      dateOfBirth: $dateOfBirth
      gender: $gender
      email: $email
      password: $password
      orgToken: $orgToken
    ) {
      user {
        profile {
          name
        }
      }
    }
  }
`;

export const GET_SIGNUP_ORGANIZATION = gql`
  query GetSignupOrganization($orgToken: String!) {
    getSignupOrganization(orgToken: $orgToken) {
      name
    }
  }
`;
