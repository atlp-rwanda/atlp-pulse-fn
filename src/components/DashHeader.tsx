import React, { useState, useEffect, useContext } from 'react';
import { Link } from 'react-router-dom';
import { MenuIcon, XIcon, SunIcon } from '@heroicons/react/outline';
import { BellIcon, MoonIcon } from '@heroicons/react/solid';
import { useLazyQuery, useSubscription, gql } from '@apollo/client';
import { toast } from 'react-toastify';
import Avatar from '../assets/avatar.png';
import Notification from './Notification';
import ProfileDropdown from './ProfileDropdown';
import { GET_PROFILE } from '../Mutations/User';
import { UserContext } from '../hook/useAuth';
import {
  getAllNotification,
  NotificationSubscription,
} from '../Mutations/notificationMutation';
import { MenuContext } from '../hook/menuProvider';
import ToggleThemeButton from './TogglethemeIcon';
import LogoIcon from './logoIcon';

export const TICKETS_NOTS_SUB = gql`
  subscription OnTicket {
    sendNotsOnTickets {
      id
      message
      createdAt
      read
      receiver

      sender {
        id
        email
        role
        profile {
          name
        }
      }
    }
  }
`;

function DashHeader() {
  /* istanbul ignore next */
  const { minimized, toggleMinMenu, showNav, toggleNav } =
    useContext(MenuContext);
  const [showNotification, setShowNotification] = useState(false);
  const [showProfileDropdown, setShowprofileDropdown] = useState(false);
  const [profileData, setProfileData] = useState<any>();
  const [getProfile] = useLazyQuery(GET_PROFILE);
  const { user, setNotificationData, addNotification } =
    useContext(UserContext);
  const [getNotification] = useLazyQuery(getAllNotification);

  const notifications = user?.notifications;

  /* istanbul ignore next */
  const handleShowNotification = () => setShowNotification(!showNotification);

  /* istanbul ignore next */
  useSubscription(TICKETS_NOTS_SUB, {
    onData: (data: any) => {
      /* istanbul ignore next */
      if (user.userId === data.data.data.sendNotsOnTickets.receiver)
        addNotification(data.data.data.sendNotsOnTickets);
    },
  });

  /* istanbul ignore next */
  useSubscription(NotificationSubscription, {
    onData: (data) => {
      /* istanbul ignore next */
      setNotificationData([data.data.data.newRating, ...notifications]);
    },
    variables: {
      receiver: user?.userId,
    },
  });

  /* istanbul ignore next */
  const handleShowProfileDropdown = () =>
    setShowprofileDropdown(!showProfileDropdown);

  /* istanbul ignore next */
  useEffect(() => {
    /* istanbul ignore next */
    const fetchData = async () => {
      try {
        const { data } = await getProfile();
        setProfileData(data);
      } catch (error: any) {
        toast.error(error?.message || 'Something went wrong');
      }
    };
    /* istanbul ignore next */
    fetchData();
  }, []);
  // eslint-disable-next-line no-nested-ternary
  useEffect(() => {
    const fetchData = async () => {
      try {
        const { data } = await getNotification();
        setNotificationData(data.getAllNotification);
      } catch (error: any) {
        // eslint-disable-next-line no-console
        console.log('error');
      }
    };
    fetchData();
  }, []);

  return (
    <>
      {showNotification && (
        <Notification handleShowNotification={handleShowNotification} />
      )}
      {showProfileDropdown && (
        <ProfileDropdown
          handleShowProfileDropdown={handleShowProfileDropdown}
        />
      )}
      <div className="w-screen py-4 z-20 bg-indigo-100 dark:bg-dark-bg page-header">
        <div className="px-3 flex items-center w-full ">
          <div className="flex px-5 lg:hidden">
            <div
              onClick={toggleNav}
              onKeyDown={toggleNav}
              role="button"
              tabIndex={0}
            >
              {!showNav ? (
                <MenuIcon className="w-7 dark:text-dark-text-fill" />
              ) : (
                <XIcon className="w-7 dark:text-dark-text-fill" />
              )}
            </div>
          </div>
          <div className="hidden lg:flex">
            <div
              onClick={toggleMinMenu}
              className="w-7"
              role="button"
              tabIndex={0}
            >
              {minimized ? (
                <MenuIcon className="w-full dark:text-dark-text-fill" />
              ) : (
                <div>
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    className="w-full box-border px-0.5"
                    style={{
                      verticalAlign: 'middle',
                      fill: 'currentColor',
                      overflow: 'hidden',
                    }}
                    version="1.1"
                    viewBox="204.77 256 563.28 460.8"
                  >
                    <path d="M445.1328 325.8368a35.3792 35.3792 0 0 1-35.84-34.9184c0-19.3024 16.0256-34.9184 35.84-34.9184H732.16c19.8144 0 35.8912 15.616 35.8912 34.9184a35.3792 35.3792 0 0 1-35.84 34.9184H445.0816z m-14.336 195.4816a35.3792 35.3792 0 0 1-35.84-34.9184c0-19.3024 16.0256-34.9184 35.84-34.9184h301.312c19.8144 0 35.8912 15.616 35.8912 34.9184a35.3792 35.3792 0 0 1-35.84 34.9184H430.7456zM283.0336 388.608l44.3392 68.7104a34.3552 34.3552 0 0 1-11.264 48.128 36.4544 36.4544 0 0 1-49.5104-11.008l-56.32-87.2448a34.0992 34.0992 0 0 1 0-37.1712l56.32-87.2448a36.4544 36.4544 0 0 1 49.4592-11.008c16.7936 10.24 21.8624 31.7952 11.264 48.128l-44.288 68.7104zM251.4432 716.8a35.3792 35.3792 0 0 1-35.84-34.9184c0-19.2512 16.0256-34.9184 35.84-34.9184h480.6656c19.8144 0 35.8912 15.6672 35.8912 34.9184a35.3792 35.3792 0 0 1-35.84 34.9184H251.392z" />
                  </svg>
                </div>
              )}
            </div>
          </div>
          <div className="flex items-center h-full lg:w-full">
            <Link to="/super-admin" className="flex flex-row lg:px-5 text-dark dark:text-dark-text-fill">
              <LogoIcon />
              <h1 className="text-3xl font-bold font-lexend">
                PULSE
              </h1>
            </Link>
          </div>

          <div className="inline-flex relative items-center p-0 text-sm font-medium text-center text-black  ml-auto dark:bg-dark-bg rounded-lg  focus:ring-4 focus:outline-none focus:ring-blue-300   dark:focus:ring-blue-800">
            <ToggleThemeButton className='dark:text-dark-text-fill' />
            <BellIcon
              className="w-6 cursor-pointer ml-auto  dark:text-dark-text-fill"
              onClick={handleShowNotification}
            />
            <span className="sr-only">Notifications</span>
            {notifications?.filter((item: any) => `${item.read}` === 'false')
              .length ? (
              <div className="inline-flex absolute -top-2 -right-2 justify-center items-center w-5 h-5 text-xs font-bold text-white bg-red-500 rounded-full border-2 border-white dark:border-gray-900">
                {
                  notifications?.filter(
                    (item: any) => `${item.read}` === 'false',
                  ).length
                }
              </div>
            ) : (
              ''
            )}
          </div>

          <div onClick={handleShowProfileDropdown}>
            <img
              className="w-8 cursor-pointer ml-4 mr-4 h-8 rounded-full"
              src={
                // eslint-disable-next-line no-nested-ternary
                user?.profileImage
                  ? // eslint-disable-next-line no-nested-ternary
                    user?.profileImage
                  : profileData?.getProfile?.avatar
                  ? // eslint-disable-next-line no-nested-ternary
                    profileData?.getProfile?.avatar
                  : Avatar
              }
              alt="avatar"
            />
          </div>
        </div>
      </div>
    </>
  );
}

export default DashHeader;
