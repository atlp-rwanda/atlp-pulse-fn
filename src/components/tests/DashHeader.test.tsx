import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import {
  gql,
  useApolloClient,
  useLazyQuery,
  useMutation,
  ApolloProvider,
  ApolloClient,
  InMemoryCache,
} from '@apollo/client';
import renderer from 'react-test-renderer';
import '../../../test/jest/__mocks__/matchMedia';
import DashHeader from '../DashHeader';

import { GET_PROFILE } from '../../Mutations/User';

const client = new ApolloClient({ cache: new InMemoryCache() });

describe('DashHeader test ', () => {
  it('Should render DashHeader', () => {
    const elem = renderer
      .create(
        <ApolloProvider client={client}>
          <BrowserRouter>
            <DashHeader />
          </BrowserRouter>
        </ApolloProvider>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
