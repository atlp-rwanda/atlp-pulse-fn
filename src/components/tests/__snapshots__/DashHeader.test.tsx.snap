// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashHeader test  Should render DashHeader 1`] = `
<div
  className="w-screen py-4 z-20 bg-indigo-100 dark:bg-dark-bg page-header"
>
  <div
    className="px-3 flex items-center w-full "
  >
    <div
      className="flex px-5 lg:hidden"
    >
      <div
        role="button"
        tabIndex={0}
      >
        <svg
          aria-hidden="true"
          className="w-7 dark:text-dark-text-fill"
          fill="none"
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </div>
    <div
      className="hidden lg:flex"
    >
      <div
        className="w-7"
        role="button"
        tabIndex={0}
      >
        <div>
          <svg
            className="w-full box-border px-0.5"
            style={
              Object {
                "fill": "currentColor",
                "overflow": "hidden",
                "verticalAlign": "middle",
              }
            }
            version="1.1"
            viewBox="204.77 256 563.28 460.8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M445.1328 325.8368a35.3792 35.3792 0 0 1-35.84-34.9184c0-19.3024 16.0256-34.9184 35.84-34.9184H732.16c19.8144 0 35.8912 15.616 35.8912 34.9184a35.3792 35.3792 0 0 1-35.84 34.9184H445.0816z m-14.336 195.4816a35.3792 35.3792 0 0 1-35.84-34.9184c0-19.3024 16.0256-34.9184 35.84-34.9184h301.312c19.8144 0 35.8912 15.616 35.8912 34.9184a35.3792 35.3792 0 0 1-35.84 34.9184H430.7456zM283.0336 388.608l44.3392 68.7104a34.3552 34.3552 0 0 1-11.264 48.128 36.4544 36.4544 0 0 1-49.5104-11.008l-56.32-87.2448a34.0992 34.0992 0 0 1 0-37.1712l56.32-87.2448a36.4544 36.4544 0 0 1 49.4592-11.008c16.7936 10.24 21.8624 31.7952 11.264 48.128l-44.288 68.7104zM251.4432 716.8a35.3792 35.3792 0 0 1-35.84-34.9184c0-19.2512 16.0256-34.9184 35.84-34.9184h480.6656c19.8144 0 35.8912 15.6672 35.8912 34.9184a35.3792 35.3792 0 0 1-35.84 34.9184H251.392z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="flex items-center h-full lg:w-full"
    >
      <a
        className="flex flex-row lg:px-5 text-dark dark:text-dark-text-fill"
        href="/super-admin"
        onClick={[Function]}
      >
        <svg
          className="w-full cursor-pointer mr-2 "
          fill="none"
          height="36"
          viewBox="0 0 59 36"
          width="59"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="9.5"
            cy="18.5"
            r="5"
            stroke="currentColor"
            strokeWidth="2"
          />
          <path
            d="M13.5 18.6112H18.2324C18.6625 18.6112 19.0444 18.3362 19.1808 17.9283L20.7981 13.0919C21.1151 12.1442 22.4728 12.1962 22.7164 13.1654L27.0802 30.5328C27.3478 31.5979 28.8876 31.5157 29.0403 30.4282L32.3788 6.64882C32.5367 5.52407 34.1498 5.49149 34.353 6.60894L38.013 26.7361C38.191 27.7149 39.5338 27.8616 39.9189 26.9442L43.1601 19.2241C43.316 18.8528 43.6794 18.6112 44.0821 18.6112H48.5"
            stroke="currentColor"
            strokeWidth="5"
          />
          <circle
            cx="52.5"
            cy="18.5"
            r="5"
            stroke="currentColor"
            strokeWidth="2"
          />
        </svg>
        <h1
          className="text-3xl font-bold font-lexend"
        >
          PULSE
        </h1>
      </a>
    </div>
    <div
      className="inline-flex relative items-center p-0 text-sm font-medium text-center text-black  ml-auto dark:bg-dark-bg rounded-lg  focus:ring-4 focus:outline-none focus:ring-blue-300   dark:focus:ring-blue-800"
    >
      <button
        className="min-w-fit px-4 cursor-pointer "
        id="theme-switch"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="w-8 dark:text-dark-text-fill"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 2a1 1 0 011 1v1a1 1 0 11-2 0V3a1 1 0 011-1zm4 8a4 4 0 11-8 0 4 4 0 018 0zm-.464 4.95l.707.707a1 1 0 001.414-1.414l-.707-.707a1 1 0 00-1.414 1.414zm2.12-10.607a1 1 0 010 1.414l-.706.707a1 1 0 11-1.414-1.414l.707-.707a1 1 0 011.414 0zM17 11a1 1 0 100-2h-1a1 1 0 100 2h1zm-7 4a1 1 0 011 1v1a1 1 0 11-2 0v-1a1 1 0 011-1zM5.05 6.464A1 1 0 106.465 5.05l-.708-.707a1 1 0 00-1.414 1.414l.707.707zm1.414 8.486l-.707.707a1 1 0 01-1.414-1.414l.707-.707a1 1 0 011.414 1.414zM4 11a1 1 0 100-2H3a1 1 0 000 2h1z"
            fillRule="evenodd"
          />
        </svg>
      </button>
      <svg
        aria-hidden="true"
        className="w-6 cursor-pointer ml-auto  dark:text-dark-text-fill"
        fill="currentColor"
        onClick={[Function]}
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 2a6 6 0 00-6 6v3.586l-.707.707A1 1 0 004 14h12a1 1 0 00.707-1.707L16 11.586V8a6 6 0 00-6-6zM10 18a3 3 0 01-3-3h6a3 3 0 01-3 3z"
        />
      </svg>
      <span
        className="sr-only"
      >
        Notifications
      </span>
    </div>
    <div
      onClick={[Function]}
    >
      <img
        alt="avatar"
        className="w-8 cursor-pointer ml-4 mr-4 h-8 rounded-full"
        src={Object {}}
      />
    </div>
  </div>
</div>
`;
